### PR TITLE
#97 Upvote and downvote feature added

### PR DESCRIPTION
**Pull request by ashwini571**

### CODE REVIEW

**Comment by ashwini571** [comment]
Unable to submit this PR link

**Comment by dbads** [comment]
> Unable to submit this PR link

What's the problem. It is working.

**Comment by ashwini571** [comment]
It's not working.The submit button is not responding.

**Comment by ashwini571** [comment]
While submissions on other issue is working

File: Projects/templates/Projects/home.html [hunk 1]
```diff
@@ -205,6 +228,68 @@ <h4>
...
+                    $(next).html('Downvotes = '+data.downvotes_count);
+
+                }
+            });
+
+
+        });
+
+
+        $('.downVote').click(function(e){
```
Comment (dbads) [inline Projects/templates/Projects/home.html]: nicely done :+1:

File: Projects/views.py [hunk 2]
```diff
@@ -8,11 +8,49 @@
...
+            issue.downvotes.remove(request.user)
+            issue.upvotes.add(request.user)
+        elif (vote_type == 'down') and (UserDownVote == 1):
+            issue.downvotes.remove(request.user)
+        else:
+            return HttpResponse('error - unknown vote type or no vote to recall')
+    upvotes_count = issue.upvotes.count()
+    downvotes_count = issue.downvotes.count()
+
+    count={'upvotes_count':upvotes_count,
```
Comment (dbads) [inline Projects/views.py]: :+1:

File: Projects/views.py [hunk 3]
```diff
@@ -8,11 +8,49 @@
...
+    UserDownVote = issue.downvotes.filter(id=request.user.id).count()
+    if (UserUpVote == 0) and (UserDownVote == 0):
+        if (vote_type == 'up'):
+            issue.upvotes.add(request.user)
+        elif (vote_type == 'down'):
+            issue.downvotes.add(request.user)
+        else:
+            return HttpResponse('error-unknown vote type')
+    else:
+        if (vote_type == 'up') and (UserUpVote == 1):
```
Comment (dbads) [inline Projects/views.py]: lots of checks :smile:

File: Projects/templates/Projects/home.html [hunk 4]
```diff
@@ -90,13 +90,33 @@ <h4>
...
+
           </div> 
           {% if request.user.profile.role != 'mentor'  %}
             <div class="md-form">
                 <input type="text" name="pr_link" id="form3" class="form-control" required="true" placeholder="PR link here">
                 <!-- <label for="form3">PR link here </label> -->
             </div>
+
             <button class="request btn btn-deep-orange"> request for verification </button>
+
```
Comment (dbads) [inline Projects/templates/Projects/home.html]: nice :+1:

**Comment by ashwini571** [comment]
Sir how do i submit the PR link??

**Pull request merged**